### ISSUE TITLE
[Bug]: Combobox clear icon(x) cannot be tabbable

### ISSUE THREAD

**Issue by Sreejith-Websphere** [carbon-design-system/carbon#18136]
### Package

@carbon/react

### Browser

Chrome

### Package version

1.54.0

### Description

During JAWS testing, we experienced that when we select an item in Combobox and if we need to clear the item using x icon, we cannot tabbed to the x icon to clear the item

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-combobox--default

### Steps to reproduce

its reproducable in carbon design documentation for combobox examples

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alina-jacob** [comment]
Hi @Sreejith-Websphere, as per our [accessibility guidelines](https://carbondesignsystem.com/components/dropdown/accessibility/), Combobox only receives a `tab` stop for the input part, the menu is navigated using `arrow` keys, opening and selection is handled with `enter`/`space`, and lastly dismissing or clearing is done using `esc`. 
So the close button would disappear upon clicking `esc` twice.

Is this what you were looking for?

**Comment by steven1046** [comment]
Hi @alina-jacob, escape twice does clear out the field for us. Thank you!
